### PR TITLE
Added cypress test for asset creation

### LINKED ISSUE #4133: Cypress tests for few functionalities
Cypress tests are to be written for the following functionalities:

- [x] #4233
- [x] #4234
- [x] #5597
- [x] #5598
- [ ] File Upload on patient consultation
- [ ] Create a new log update on the consultation page
- [ ] update existing log update
- [x] #4235
- [x] create a new asset for a facility (PR: #4134 )
- [ ] configuration of the asset page
- [x] #4320
- [x] #4321

Google Sheets Link: https://docs.google.com/spreadsheets/d/1S77kcGMjVRgVlm9lA0WL2iV_GEhxs9xocm3ZD7p4uCY/

### PR DESCRIPTION
**Pull request by ShivamJhaa**
### WHAT

Added cypress test for asset creation

## Proposed Changes

- Fixes #4133 

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### <samp>🤖 Generated by Copilot at dec2ae0</samp>

*  Add `data-testid` attributes to various input fields and buttons in `AssetCreate.tsx` and `AssetsList.tsx` to enable locating them in test cases ([link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d61b507f07891c695d44e81effbf8f1da3a8667b0379dcfbd98c5761aa124f49L369-R372), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073L513-R517), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073L528-R536), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073L543-R555), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073L572-R588), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073L594-R613), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073L605-R627), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073L616-R642), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073L669-R695), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073R721), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073R738), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073R768), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073R785), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073R801), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073R820), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073R837), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073R856), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-d7d19aab87f2678510fa29d56b44e2a94098c956b954df68110c117f7f366073L850-R888))
*  Add a test case for creating an asset in `assets.cy.ts` that fills in the required fields and verifies the success notification ([link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-5979985dda75f49bd8934fbd4aff8bdb17579a17a3fb07a02119011a57c335e2R17-R73))
*  Import the `uuid` module in `assets.cy.ts` and `vite.config.ts` to generate unique identifiers for the asset QR codes in the test cases ([link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-5979985dda75f49bd8934fbd4aff8bdb17579a17a3fb07a02119011a57c335e2L4-R6), [link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-6a3b01ba97829c9566ef2d8dc466ffcffb4bdac08706d3d6319e42e0aa6890ddR60))
*  Add a comment to disable the eslint rule for using `require` in `vite.config.ts` ([link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-6a3b01ba97829c9566ef2d8dc466ffcffb4bdac08706d3d6319e42e0aa6890ddR60))
*  Remove an empty line in `patient_crud.cy.ts` ([link](https://github.com/coronasafe/care_fe/pull/5632/files?diff=unified&w=0#diff-85edc6c535fd4c7cef250d5a68bb562c3fd75a82654e2de0b0227413d5840bfaR159))

### CODE REVIEW

**Comment by ShivamJhaa** [comment]
@nihal467 @khavinshankar Please review the PR.
Thanks!

File: cypress/e2e/assets_spec/assets.cy.ts [hunk 1]
```diff
@@ -13,6 +14,63 @@ describe("Assets List", () => {
     cy.awaitUrl("/assets");
   });
 
+  it("Create an Asset", () => {
+    cy.get("button").should("contain", "Create Asset");
+    cy.get("[data-testid=create-asset-buttom] button").click();
+    cy.get("input[name='facilities']")
+      .type("Govt Hosp, Ernakulam")
```
Comment (nihal467) [inline cypress/e2e/assets_spec/assets.cy.ts]:
```suggestion
      .type("Dummy Facility 1")
```

File: cypress/e2e/assets_spec/assets.cy.ts [hunk 2]
```diff
@@ -13,6 +14,63 @@ describe("Assets List", () => {
...
+      .then(() => {
+        cy.get("[role='option']").contains("Govt Hosp, Ernakulam").click();
```
Comment (nihal467) [inline cypress/e2e/assets_spec/assets.cy.ts:23]:
```suggestion
        cy.get("[role='option']").contains("Dummy Facility 1").click();
```

Comment (nihal467) [inline cypress/e2e/assets_spec/assets.cy.ts]:
```suggestion
        cy.get("[role='option']").contains("Dummy Facility 1").click();
```

**Comment by nihal467** [comment]
@ShivamJhaa The cypress test run by the GitHub action uses a dedicated backend, with dummy data loaded to it, it is only having facilities named "Dummy Facility 1" and "Dummy Shifting Center". Currently, I have updated it, but next time make sure you use those facilities only :)

**Comment by nihal467** [comment]
@ShivamJhaa there is a failure in cypress, check it

**Comment by ShivamJhaa** [comment]
> @ShivamJhaa The cypress test run by the GitHub action uses a dedicated backend, with dummy data loaded to it, it is only having facilities named "Dummy Facility 1" and "Dummy Shifting Center". Currently, I have updated it, but next time make sure you use those facilities only :)

Strangely, the test was passing on Cl even when I was using the local backend data.
https://github.com/coronasafe/care_fe/actions/runs/5215017623/jobs/9412022568#step:10:66

**Comment by ShivamJhaa** [comment]
@nihal467 Please review it, the tests are passing now.
Thanks!

File: vite.config.ts [hunk 3]
```diff
@@ -82,7 +83,7 @@ export default defineConfig({
     port: 4000,
     proxy: {
       "/api": {
-        target: "https://careapi.ohc.network",
+        target: "http://localhost:9000",
```
Comment (nihal467) [inline vite.config.ts]:
```suggestion
        target: "https://careapi.ohc.network",
```

File: vite.config.ts [hunk 4]
```diff
@@ -73,7 +74,7 @@ export default defineConfig({
...
```
Comment (nihal467) [inline vite.config.ts]:
```suggestion
        target: "https://careapi.ohc.network",
```

**Review by nihal467 — CHANGES_REQUESTED** [review]
before pushing the PR, make sure that you revert the changes you are making in vite.config.ts to run your own backend

File: vite.config.ts (hunk 4 above)
Comment (ShivamJhaa) [reply vite.config.ts]: Done.

**Comment by ShivamJhaa** [comment]
> @ShivamJhaa before pushing the PR, make sure that you revert the changes you are making in vite.config.ts to run your own backend

Sure, will keep this in mind, thanks!

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Comment by khavinshankar** [comment]
@nihal467 can u review and approve this pr

**Review by nihal467 — APPROVED** [review]

**Pull request merged**